compiler/qsc_parse/src/lex/cooked.rs

```diff
@@ -357,15 +380,25 @@ impl<'a> Lexer<'a> {
     }
 
     #[allow(clippy::too_many_lines)]
-    fn single(&mut self, single: Single) -> Result<TokenKind, Error> {
+    fn single(&mut self, single: Single, span: Span) -> Result<TokenKind, Error> {
```

Comment:
`single` doesn't need this parameter. The other incomplete errors use the span of the next token, not the span of the current token.